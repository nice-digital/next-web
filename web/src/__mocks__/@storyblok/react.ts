export const storyblokInit = (): null => null;

export const apiPlugin = (): null => null;
