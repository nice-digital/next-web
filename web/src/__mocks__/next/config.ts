import nextConfig from "./../../../next.config";

const getConfig = (): unknown => ({
	publicRuntimeConfig: nextConfig.publicRuntimeConfig,
	serverRuntimeConfig: nextConfig.serverRuntimeConfig,
});

export default getConfig;
