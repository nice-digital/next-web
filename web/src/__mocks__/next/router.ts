export const useRouter = jest.fn().mockImplementation(() => {
	throw new Error("Provide a mock of useRouter for testing purposes");
});
