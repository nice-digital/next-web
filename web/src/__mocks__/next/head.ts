/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import { FC } from "react";
import { renderToString } from "react-dom/server";

interface HeadProps {
	children: Parameters<typeof renderToString>[0];
}

/**
 * Mocking next/head allows us to use it in our tests to assert against meta tags/titles
 * See https://edibleco.de/3dyJuUl
 */
const Head: FC<HeadProps> = ({ children }: HeadProps) => {
	if (children) {
		global.document.head.insertAdjacentHTML(
			"afterbegin",
			renderToString(children) || ""
		);
	}
	return null;
};

export default Head;
