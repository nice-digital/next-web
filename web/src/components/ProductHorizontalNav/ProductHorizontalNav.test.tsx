import { render, screen } from "@testing-library/react";
import { useRouter } from "next/router";

import {
	ProductHorizontalNav,
	type ProductHorizontalNavProps,
} from "./ProductHorizontalNav";

const props: ProductHorizontalNavProps = {
	productTypeName: "Indicator",
	productPath: "/indicators/ind1-test",
	hasEvidenceResources: true,
	hasHistory: true,
	hasInfoForPublicResources: true,
	hasToolsAndResources: true,
};

const useRouterMock = jest.mocked(useRouter);

describe("ProductHorizontalNav", () => {
	beforeEach(() => {
		useRouterMock.mockReturnValue({
			asPath: props.productPath,
		} as ReturnType<typeof useRouter>);
	});

	it("should match snapshot for all nav items", () => {
		const { container } = render(<ProductHorizontalNav {...props} />);
		expect(container).toMatchSnapshot();
	});

	it("should use product type name for overview nav link", () => {
		render(<ProductHorizontalNav {...props} />);
		expect(screen.getByRole("link", { name: "Indicator" })).toBeInTheDocument();
	});

	it("should highlight overview nav link when on the overview page", () => {
		render(<ProductHorizontalNav {...props} />);
		expect(screen.getByRole("link", { name: "Indicator" })).toHaveAttribute(
			"aria-current",
			"true"
		);
	});

	it("should render nothing when there are no resources, IFP, evidence or history", () => {
		render(
			<ProductHorizontalNav
				{...props}
				hasToolsAndResources={false}
				hasInfoForPublicResources={false}
				hasEvidenceResources={false}
				hasHistory={false}
			/>
		);

		expect(screen.queryByRole("navigation")).toBeNull();
	});

	it.each([
		["Tools and resources", "hasToolsAndResources"],
		["Information for the public", "hasInfoForPublicResources"],
		["Evidence", "hasEvidenceResources"],
		["History", "hasHistory"],
	])("should hide %s link when no content", (name, propName) => {
		const newProps = { ...props, [propName]: false };

		render(<ProductHorizontalNav {...newProps} />);

		expect(screen.queryByRole("link", { name })).toBeNull();
	});

	it.each([
		["Tools and resources", "resources"],
		["Information for the public", "information-for-the-public"],
		["Evidence", "evidence"],
		["History", "history"],
	])("should highlight %s link when exact match on path", (name, pathSlug) => {
		useRouterMock.mockReturnValue({
			asPath: `${props.productPath}/${pathSlug}`,
		} as ReturnType<typeof useRouter>);

		render(<ProductHorizontalNav {...props} />);

		expect(screen.getByRole("link", { name })).toHaveAttribute(
			"aria-current",
			"true"
		);
	});

	it.each([
		["Tools and resources", "resources"],
		["Information for the public", "information-for-the-public"],
		["Evidence", "evidence"],
		["History", "history"],
	])("should highlight %s link for sub pages", (name, pathSlug) => {
		useRouterMock.mockReturnValue({
			asPath: `${props.productPath}/${pathSlug}/sub-page`,
		} as ReturnType<typeof useRouter>);

		render(<ProductHorizontalNav {...props} />);

		expect(screen.getByRole("link", { name })).toHaveAttribute(
			"aria-current",
			"true"
		);
	});
});
