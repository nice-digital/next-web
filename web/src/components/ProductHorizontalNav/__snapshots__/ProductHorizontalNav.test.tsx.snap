// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductHorizontalNav should match snapshot for all nav items 1`] = `
<div>
  <nav
    class="horizontal-nav"
  >
    <ul
      class="horizontal-nav__list"
    >
      <li
        class="horizontal-nav__item"
      >
        <a
          aria-current="true"
          class="horizontal-nav__link"
          href="/indicators/ind1-test"
        >
          Indicator
        </a>
      </li>
      <li
        class="horizontal-nav__item"
      >
        <a
          aria-current="false"
          class="horizontal-nav__link"
          href="/indicators/ind1-test/resources"
        >
          Tools and resources
        </a>
      </li>
      <li
        class="horizontal-nav__item"
      >
        <a
          aria-current="false"
          class="horizontal-nav__link"
          href="/indicators/ind1-test/evidence"
        >
          Evidence
        </a>
      </li>
      <li
        class="horizontal-nav__item"
      >
        <a
          aria-current="false"
          class="horizontal-nav__link"
          href="/indicators/ind1-test/history"
        >
          History
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
